runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7427,10 +7496,12 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
 #endif /* defined(J9VM_OPT_JITSERVER) */
          _vm = TR_J9VMBase::get(_jitConfig, vmThread, TR_J9VMBase::AOT_VM);
 #if defined(J9VM_OPT_JITSERVER)
-      if ((_compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT) &&
-         !shouldPerformLocalComp(entry))
+      if (_compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT)
          {
-         entry->setRemoteCompReq();
+         if (!shouldPerformLocalComp(entry))
+            entry->setRemoteCompReq();
+         else
+            downgradeLocalCompilationIfLowPhysicalMemory(entry);
```

Comment:
Sometimes we want to perform cheap compilation locally to avoid networking latency. Such compilations should not be added to LPQ, but they should go through the normal recompilation mechanisms because they were not downgraded due to server not being available.
I think we should distinguish between different reasons that compilation are performed locally.